src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt

```diff
@@ -222,8 +226,14 @@ open class IndexReplicationTask(id: Long, type: String, action: String, descript
                                     updateMetadata()
                                 }
                                 is FailedState -> {
-                                    // Try pausing first if we get Failed state. This returns failed state if pause failed
-                                    pauseReplication(state)
+                                    // Try pausing or stopping if we get Failed state based on settings.
+                                    // This returns failed state if pause or stop failed
+                                    if (clusterService.clusterSettings.get(ReplicationPlugin.REPLICATION_REPLICATE_INDEX_DELETION)
+                                        && clusterService.state().metadata.index(leaderIndex) == null) {
```

Comment:
Why are we checking for leader index in follower's cluster state? shouldn't it be checked on leader's state? 
Other option could be to check the reason for failure and use that to trigger follower index delete.

Reply:
Changed the logic to check the failure reason is "IndexNotFoundException of leader index" .